pandas/io/excel.py

```diff
@@ -207,6 +210,18 @@ def __init__(self, io, **kwds):
         if engine is not None and engine != 'xlrd':
             raise ValueError("Unknown engine: %s" % engine)
 
+        # io can be a path object but relevant libraries may not be
+        # installed
+        # get_filepath_or_buffer() converts path objects to string
+        if _PATHLIB_INSTALLED:
+            from pathlib import Path
+            if isinstance(io, Path):
+                io = get_filepath_or_buffer(io)[0]
+        if _PY_PATH_INSTALLED:
+            from py.path import local as LocalPath
+            if isinstance(io, LocalPath):
```

Comment:
you don't need any of this
just call get_filepath_or_buffer (actually can prob eliminate almost all of the s3 checking in this function - this is all handled in that function)